docs/css/style.css

```diff
@@ -1,10 +1,123 @@
 body {
   font-family: 'Open Sans', Helvetica, Arial, FreeSans;
-  color: #333;
+  color: var(--text-secondary, #333);
   -webkit-font-smoothing: antialiased;
   -webkit-text-size-adjust: 100%;
   padding: 0;
   margin: 0;
+  background-color: var(--bg-primary, #ffffff);
+  transition: background-color 0.3s ease, color 0.3s ease;
+}
+
+/* Dark mode support for homepage */
+@media (prefers-color-scheme: dark) {
+  body:not([data-theme="light"]) {
+    background-color: var(--bg-primary, #1a1a1a);
+    color: var(--text-secondary, #c0c0c0);
+  }
+}
+
+/* Manual theme overrides */
+body[data-theme="dark"] {
+  background-color: var(--bg-primary, #1a1a1a);
+  color: var(--text-secondary, #c0c0c0);
+}
+
+body[data-theme="light"] {
+  background-color: var(--bg-primary, #ffffff);
+  color: var(--text-secondary, #333);
+}
+
+/* Theme Toggle Button for Homepage */
+#theme-toggle {
+  position: fixed;
+  top: 10px;
```

Comment:
Putting the theme toggle on the upper left seems to conflict with the logo - better to move to lower right WDYT?

<img width="248" height="138" alt="image" src="https://github.com/user-attachments/assets/121a46ad-5214-488e-ae36-b2576284ba9b" />